source/server/connection_handler_impl.cc

```diff
@@ -327,27 +333,28 @@ ConnectionHandlerImpl::ActiveConnection::ActiveConnection(ActiveTcpListener& lis
   connection_->addConnectionCallbacks(*this);
   listener_.stats_.downstream_cx_total_.inc();
   listener_.stats_.downstream_cx_active_.inc();
+  listener_.per_worker_stats_.downstream_cx_total_.inc();
+  listener_.per_worker_stats_.downstream_cx_active_.inc();
 }
 
 ConnectionHandlerImpl::ActiveConnection::~ActiveConnection() {
   listener_.stats_.downstream_cx_active_.dec();
   listener_.stats_.downstream_cx_destroy_.inc();
+  listener_.per_worker_stats_.downstream_cx_active_.dec();
   conn_length_->complete();
 }
 
-ListenerStats ConnectionHandlerImpl::generateStats(Stats::Scope& scope) {
-  return {ALL_LISTENER_STATS(POOL_COUNTER(scope), POOL_GAUGE(scope), POOL_HISTOGRAM(scope))};
-}
-
-ActiveUdpListener::ActiveUdpListener(Event::Dispatcher& dispatcher, Network::ListenerConfig& config)
-    : ActiveUdpListener(dispatcher.createUdpListener(config.socket(), *this), config) {}
+ActiveUdpListener::ActiveUdpListener(Network::ConnectionHandler& parent,
+                                     Event::Dispatcher& dispatcher, Network::ListenerConfig& config)
+    : ActiveUdpListener(parent, dispatcher.createUdpListener(config.socket(), *this), config) {}
 
-ActiveUdpListener::ActiveUdpListener(Network::ListenerPtr&& listener,
+ActiveUdpListener::ActiveUdpListener(Network::ConnectionHandler& parent,
+                                     Network::ListenerPtr&& listener,
                                      Network::ListenerConfig& config)
-    : ConnectionHandlerImpl::ActiveListenerImplBase(std::move(listener), config),
-      udp_listener_(dynamic_cast<Network::UdpListener*>(listener_.get())), read_filter_(nullptr) {
+    : ConnectionHandlerImpl::ActiveListenerImplBase(parent, std::move(listener), config),
+      udp_listener_(*dynamic_cast<Network::UdpListener*>(listener_.get())), read_filter_(nullptr) {
```

Comment:
I think this part is not synced with #7896

Reply:
It is up to date. I just, switched it to a reference since it can't be null.